modules/web/js/ballerina/diagram2/views/default/components/decorators/panel-decorator.jsx

```diff
@@ -80,6 +81,18 @@ class PanelDecorator extends React.Component {
         this.props.model.remove();
     }
 
+    togglePublicPrivateFlag() {
+        this.props.model.public = !this.props.model.public;
+        this.props.model.trigger('tree-modified', {
```

Comment:
If its to update the view, we can simply call `this.context.editor.update();` and have context types:
```
PanelDecorator.contextTypes = {
    editor: PropTypes.instanceOf(Object).isRequired,
};
```

Reply:
We have to update the view and should do a source-gen as well. That is the reason a tree modification is done so it updates the view as well as do a source-gen.

Reply:
👍